posthog/api/feature_flag.py

```diff
@@ -157,12 +158,14 @@ def properties_all_match(predicate):
                 prop = Property(**property)
                 if prop.type == "cohort":
                     try:
-                        cohort: Cohort = Cohort.objects.get(pk=prop.value, team_id=self.context["team_id"])
-                        if [prop for prop in cohort.properties.flat if prop.type == "behavioral"]:
-                            raise serializers.ValidationError(
-                                detail=f"Cohort '{cohort.name}' with behavioral filters cannot be used in feature flags.",
-                                code="behavioral_cohort_found",
-                            )
+                        initial_cohort: Cohort = Cohort.objects.get(pk=prop.value, team_id=self.context["team_id"])
+                        dependent_cohorts = get_dependent_cohorts(initial_cohort)
+                        for cohort in [initial_cohort, *dependent_cohorts]:
+                            if [prop for prop in cohort.properties.flat if prop.type == "behavioral"]:
+                                raise serializers.ValidationError(
+                                    detail=f"Cohort '{cohort.name}' with behavioral filters cannot be used in feature flags.",
```

Comment:
In the UI does this just raise an error when you select the cohort while you're editing? Or you have to save before it errors you out?

Also might be nice to just disable behavioral cohorts in the filters selection in the FE then so people can easily tell they can't pick those

Reply:
When you save. Agree, the UX here isn't ideal; we've never gotten around to it because low priority.